packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -877,6 +877,16 @@ export interface VpcProps {
    */
   readonly maxAzs?: number;
 
+  /**
+   * Define the number of AZs to reserve.
+   *
+   * When specified, the IP space is reserved for the azs but no actual
+   * resources are provisioned.
+   *
+   * @default 0
+   */
+  readonly reservedAzs?: number;
```

Comment:
I'm not really sure that this is the right fix here. After reading through the comments on the issue, it's clear that we do need a fix here, but how often will people think ahead to reserve additional AZs?

I think our team needs to discuss a path forward on this one so, full disclaimer, this is where I'm stopping this review for now. I don't want to waste your time in edits when I'm just not sure this is the right path forward.